tests/behat/features/bootstrap/DataManager.php

```diff
@@ -0,0 +1,138 @@
+<?php
+
+/**
+ * @file
+ * DataManager. Handles data from files.
+ */
+
+/**
+ * Class DataManager
+ */
+class DataManager extends \Page\PageBase {
+  /**
+   * Filename.
+   *
+   * @var string $filename
+   */
+  protected $filename = '';
+
+  /**
+   * Flag for reservable only.
+   *
+   * @var bool $onlyReservable
+   *    True if only finding reservable objects.
+   */
+  protected $onlyReservable = false;
+
+  /**
+   * GetFilename.
+   *
+   * @return mixed
+   *    The current filename.
+   */
+  public function getFilename() {
+    return $this->filename;
+  }
+
+  /**
+   * GetRandomPID
```

Comment:
What/why does this do

Reply:
Explained in comments now. Essentially we're using equivalence classes of data instead of fixed data examples. This gives the advantage over time to cover more test data examples. Maybe for CCI tests the datafile should only have one, so it is more deterministic, but it shouldn't be a problem unless there is data examples the system cannot handle (although it should).